### ISSUE TITLE
why lables same with input_ids?

### ISSUE THREAD

**Issue by candowu** [tloen/alpaca-lora#365]
in  def tokenize(prompt, add_eos_token=True): function, i find this:

result["labels"] = result["input_ids"].copy()

why labels same as input_ids?
What are the benefits of doing this? Why not use the output tokenize result?

**Comment by KukumavMozolo** [comment]
When you look into LlamaForCausalLM, in the forward method you will find  the following code:

```
        if labels is not None:
            # Shift so that tokens < n predict n
            shift_logits = logits[..., :-1, :].contiguous()
            shift_labels = labels[..., 1:].contiguous()
```

So as one can see the logits are shifted backwards by 1(last element is droped) and in the labels the first element is droped.

Therefore the logits that are computed based on the first input token. This when feed into a softmax will correspond to a probability distribution over the possible second token and since the labels are shifted this will be compared to the true value for the second token.
